python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -297,7 +298,7 @@ def _address_family_evpn(self: AvdStructuredConfigOverlay) -> dict:
             }
             address_family_evpn["neighbors"] = [{"ip_address": self._wan_ha_peer_vtep_ip(), "activate": True}]
 
-        return address_family_evpn
+        return address_family_evpn or None
```

Comment:
need to adjust the return type in the function signature - but do we really want to return None rather than an empty dict as we tend to do everywhere else? Usually we would expect the caller to not render empty (strip_empty..)

Reply:
toh

Ok we are doing this in main.. so not removing empty dicts. sad sad
```
       # Need to keep potentially empty dict for redistribute_routes
        return strip_empties_from_dict(router_bgp, strip_values_tuple=(None, ""))
```

let's change the signature then

Reply:
fixed

Reply:
fixed